smtpd.go

```diff
@@ -475,7 +475,12 @@ loop:
 			if s.srv.Handler != nil {
 				err := s.srv.Handler(s.conn.RemoteAddr(), from, to, buffer.Bytes())
 				if err != nil {
-					s.writef("451 4.3.5 Unable to process mail")
+					checkErrFormat := regexp.MustCompile(`^([2-5][0-9]{2})[\s\-](.+)$`)
+					if checkErrFormat.MatchString(err.Error()) {
+						s.writef(err.Error())
```

Comment:
This breaks RFC compliance by not including the error code at the start. Perhaps the error detail could be supplied after the "Unable to process mail" string?

Reply:
There is regexp check for proper format for RFC compliance. This is useful feature if you need to provide custom error code and message.